src/librustc/ty/context.rs

```diff
@@ -207,52 +207,55 @@ pub struct CommonTypes<'tcx> {
 
 #[derive(RustcEncodable, RustcDecodable)]
 pub struct TypeckTables<'tcx> {
+    /// The HirId::owner all ItemLocalIds in this table are relative to.
+    pub local_id_root: DefId,
```

Comment:
This is in several places initialized with an `invalid_def_id`. Why can't this be an `Option`?